core/src/main/java/org/elasticsearch/index/mapper/core/StringFieldMapper.java

```diff
@@ -300,6 +301,14 @@ protected void parseCreateField(ParseContext context, List<Field> fields) throws
         }
     }
 
+    private int lengthInBytes(final String str) {
+        try {
+            return str.getBytes("UTF-8").length;
+        } catch (UnsupportedEncodingException e) {
```

Comment:
Please use String.getBytes(Charset) and pass StandardCharsets.UTF_8 for type safety and better performance.